### ISSUE TITLE
Prettier support

### ISSUE THREAD

**Issue by rauchg** [zeit/styled-jsx#222]
Would be amazing to find a way to make prettier work within `<style jsx>` tags :)

https://twitter.com/Vjeux/status/870670634908475392

**Comment by rauchg** [comment]
Issue to track:
https://github.com/prettier/prettier/issues/1948

**Comment by giuseppeg** [comment]
It should be fairly easy to do I think. Probably the best way to get it done would be to contribute to Prettier. Another option would be create prettier-styled-jsx when support for plugins (#190) ships.

**Comment by pomber** [comment]
Basic support for styled-jsx was merged today with prettier/prettier#1984.  
Next step will be to format template literals containing expressions. Any tricky test case that you can think of?

**Comment by giuseppeg** [comment]
@pomber you can see the fixture from this PR https://github.com/zeit/styled-jsx/pull/227

**Comment by giuseppeg** [comment]
Prettier `^1.5.0` formats styled-jsx but ignores our "external styles" because they are just exported strings / template literals eg.

```js
export button = `
  button { color: red }
`
```

In styled-jsx we detect external styles by validating exported strings / template literals with CSSTree [[relevant code](https://github.com/zeit/styled-jsx/blob/master/src/_utils.js#L194-L242)]

@vjeux would you accept a PR to do the same in Prettier? Maybe the overhead is minimal.

**Comment by vjeux** [comment]
I'm not comfortable with this option. The CSSTree spec is extremely lenient and many things that aren't CSS can parse as CSS.

I would encourage you to add a way to statically tag a template literal as being css. Something like

```js
import { css } from 'styled';

export default css`
`
```

It's definitely more annoying and verbose but I believe it's going to be better for the ecosystem in general.

**Comment by giuseppeg** [comment]
@vjeux fwiw I found CSSTree being rather strict and more close to the w3c validator than say PostCSS.

That said I agree with you that tagging the template or annotating the css string with a comment is the best way to go.

**Comment by giuseppeg** [comment]
@rauchg @nkzawa let's keep this in mind and include the change in the next major release, whenever that will be.

**Comment by vjeux** [comment]
Oh thanks for the mention, I was indeed talking about postcss. I've never used CSSTree

**Comment by giuseppeg** [comment]
Fixed by #288

**Comment by awitherow** [comment]
Doesn't seem to work by default within my project as of now...  see top

```
<style jsx>{`
  
            @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,700i&display=swap');
            
            :global(body) {
              margin: 0;
              font-family: "Open Sans", -apple-system, BlinkMacSystemFont, Avenir Next, Avenir,
                Helvetica, sans-serif;
              font-weight: 300;
  
              background: url("${require('../public/img.jpg')}") no-repeat center center fixed; 
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
            }
            h1, h2, h3, h4, h5, h6 {
              font-weight: 700;
              font-style: italic;
            }
  
            .page {
              display: flex;
              align-items: center;
              justify-content: center;
              min-height: 100vh;
            }
        .hero {
          max-width: 520px;
          margin: 4rem 1rem;
        }
        .title {
          margin: 0;
          width: 100%;
          line-height: 1.15;
          font-size: 48px;
          padding: 0.33rem;
        }
        .description {
          font-style: unset !important;
        }
        
        .row {
          max-width: 880px;
          margin: 80px auto 40px;
          display: flex;
          flex-direction: row;
          justify-content: space-around;
        }
        .card {
          padding: 18px 18px 24px;
          width: 220px;
          text-align: left;
          text-decoration: none;
          color: #434343;
          border: 1px solid #9b9b9b;
        }
        .card:hover {
          border-color: #067df7;
        }
        .card h3 {
          margin: 0;
          color: #067df7;
          font-size: 18px;
        }
        .card p {
          margin: 0;
          padding: 12px 0 0;
          font-size: 13px;
          color: #333;
        }
      `}</style>
```
